reference-project/src/main/java/com/devonfw/demoquarkus/rest/v1/controller/SecureRestController.java

```diff
@@ -0,0 +1,85 @@
+package com.devonfw.demoquarkus.rest.v1.controller;
+
+import org.eclipse.microprofile.jwt.JsonWebToken;
+import javax.annotation.security.PermitAll;
+import javax.annotation.security.RolesAllowed;
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+import javax.ws.rs.*;
+import javax.ws.rs.core.*;
+
+@RequestScoped
+@Path("/secure")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class SecureRestController {
+    
+    @Inject
+    JsonWebToken jwt;
+    
+    @GET()
+    @Path("/permit")
+    @PermitAll 
+    @Produces(MediaType.TEXT_PLAIN)
+    public String permit() {
+        return "Access to /secure/permit was granted!";
+    }
+    
+    @GET()
+    @Path("/token")
+    @PermitAll
+    @Produces(MediaType.TEXT_PLAIN)
+    public String token(@Context SecurityContext ctx) {
+        String name;
+        if (ctx.getUserPrincipal() == null) { 
+            name = "anonymous";
+        } else if (!ctx.getUserPrincipal().getName().equals(jwt.getName())) { 
+            throw new InternalServerErrorException("Principal and JsonWebToken names do not match");
+        } else {
+            name = ctx.getUserPrincipal().getName(); 
+        }
+        
+        StringBuilder s = new StringBuilder();
+        if(jwt != null && ctx != null) {
+        	s.append(String.format("hello + %s,"
+                    + " isHttps: %s,"
+                    + " authScheme: %s,",
+                    name, ctx.isSecure(), ctx.getAuthenticationScheme()));
+            s.append(String.format("issuer: %s\n"
+            		+ "tokenID: %s\n"
+            		+ "rawToken: %s",
+            		jwt.getIssuer(), jwt.getTokenID(), jwt.getRawToken()));
+            if(jwt.getClaimNames() != null) {
+            	for(String claim : jwt.getClaimNames()) {
+                	s.append(claim + ": " + jwt.getClaim(claim).toString() + "\n");
+                }
+            }
+        }
+        return s.toString(); 
+    }
+    
+    @GET
+    @Path("/roles-allowed/admin") 
+    @RolesAllowed({ "admin" }) 
```

Comment:
OK. So like in spring we have the freedom to put `@RolesAllowed` annotation "everywhere" and it is just working via AOP. So next we could think about some best practices how to do it.
In devon4j we always annotated use-cases as one use-case may call another use-case.
Do we give up such principles in quarkus and let the users do whatever they like?
Will we then call functionality from one service method in another service method? What if `@Context` parameters are required that prevent such usage.
My experience was that the REST service is simply exposing functionality to the outside world but nothing more.
However, I am already discussing beyond the goal of this acutal PR...